### PR TITLE
Pass the model appends in the new intances

### PR DESCRIPTION
**Pull request by ajcastro**
I think it is good if we also pass the appends so that we can set appends before querying the models something like:
```php
(new User)->append(['full_name', 'age'])->get(); // this will return users with appends of full_name and age
```

Please forgive me if I am not sure if I created this PR correctly, please let me know if something needs here to be corrected

### CODE REVIEW

**Comment by taylorotwell** [comment]
This would probably be considered a breaking change so would have to be considered on the master branch.

**Comment by ajcastro** [comment]
Hi @taylorotwell. Yes sorry the `appends` really break and I experienced it now in my local. I should have use `setAppends()` instead, and I achieve the behavior I intended. Do you think it is good if we change this to `setAppends()` or it is still possible that it will be a breaking change? For the meantime, I just override the method `newInstance` in my models and add `$model->setAppends($this->appends)` there.

**Comment by ajcastro** [comment]
Hi @taylorotwell. Currently, there are two ways to override the default appends.

1. Create another model class extending to the class to override the `$appends` property.
```php
class UserWithAppends extends App\User {
  protected $appends = [ 'full_name', 'age' ];
}
```

2. Call setAppends on retrieved models
```php
public function index() {
  return User::get()->map(function($user) {
    return $user->setAppends([ 'full_name', 'age' ]);
  });
}
```

If we call the `setAppends()` in the `Illuminate\Database\Eloquent\Model::newInstance()`, we can achieve it like this:
```php
(new User)->append(['full_name', 'age'])->get();
```

what are your thoughts on this?

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -412,6 +412,8 @@ public function newInstance($attributes = [], $exists = false)
         $model->setTable($this->getTable());
 
         $model->mergeCasts($this->casts);
+        
+        $model->append($this->appends);
```
Comment (ajcastro) [inline src/Illuminate/Database/Eloquent/Model.php:416]: should be   `$model->setAppends($this->appends);`

**Pull request closed without merging**